contracts/vaults/BaseVaults/RibbonThetaVault.sol

```diff
@@ -111,57 +143,65 @@ contract RibbonThetaVault is RibbonVault, RibbonThetaVaultStorage {
-       black-scholes premium calculation logic
-     * @param _strikeSelection is the address of the contract with strike selection logic
-     * @param _premiumDiscount is the vault's discount applied to the premium
-     * @param _auctionDuration is the duration of the gnosis auction
-     * @param _vaultParams is the struct with vault general data
-     */
+    // /**
+    //  * @notice Initializes the OptionVault contract with storage variables.
+    //  * @param params is the struct with initialization parameters
+    //  * @param _vaultParams is the struct with vault general data
+    //  */
     function initialize(
-        address _owner,
-        address _keeper,
-        address _feeRecipient,
-        uint256 _managementFee,
-        uint256 _performanceFee,
-        string memory _tokenName,
-        string memory _tokenSymbol,
-        address _optionsPremiumPricer,
-        address _strikeSelection,
-        uint32 _premiumDiscount,
-        uint256 _auctionDuration,
+        initParams memory params,
         Vault.VaultParams calldata _vaultParams
     ) external initializer {
         baseInitialize(
-            _owner,
-            _keeper,
-            _feeRecipient,
-            _managementFee,
-            _performanceFee,
-            _tokenName,
-            _tokenSymbol,
+            params._owner,
+            params._keeper,
+            params._feeRecipient,
+            params._managementFee,
+            params._performanceFee,
+            params._tokenName,
+            params._tokenSymbol,
             _vaultParams
         );
-        require(_optionsPremiumPricer != address(0), "!_optionsPremiumPricer");
-        require(_strikeSelection != address(0), "!_strikeSelection");
         require(
-            _premiumDiscount > 0 &&
-                _premiumDiscount < 100 * Vault.PREMIUM_DISCOUNT_MULTIPLIER,
+            params._optionsPremiumPricer != address(0),
+            "!_optionsPremiumPricer"
+        );
+        require(params._strikeSelection != address(0), "!_strikeSelection");
+        require(
+            params._premiumDiscount > 0 &&
+                params._premiumDiscount <
+                100 * Vault.PREMIUM_DISCOUNT_MULTIPLIER,
             "!_premiumDiscount"
         );
-        require(_auctionDuration >= MIN_AUCTION_DURATION, "!_auctionDuration");
-        optionsPremiumPricer = _optionsPremiumPricer;
-        strikeSelection = _strikeSelection;
-        premiumDiscount = _premiumDiscount;
-        auctionDuration = _auctionDuration;
+        require(
+            params._auctionDuration >= MIN_AUCTION_DURATION,
+            "!_auctionDuration"
+        );
+        optionsPremiumPricer = params._optionsPremiumPricer;
+        strikeSelection = params._strikeSelection;
+        premiumDiscount = params._premiumDiscount;
+        auctionDuration = params._auctionDuration;
+
+        usdcAuction = params._usdcAuction;
+        if (usdcAuction) {
```

Comment:
This should just be `if (params._usdcAuction)` because reading `usdcAuction` costs more gas (reading from storage)

Reply:
Fixed